ideascube/utils.py

```diff
@@ -47,3 +47,8 @@ def to_unicode(text):
         except UnicodeDecodeError:
             text = text.decode('latin-1')
         return text
+
+
+def iter_file(file):
+    for row in file:
+        yield to_unicode(row)
```

Comment:
> Sorry, I don't get why it does not work. Aren't we decoding line by line?

And that's exactly the problem, as I mentioned in my previous comment.

Say the file is in the Latin-1 encoding, but the first line decodes fine as UTF-8.

Your iterator will yield the first line decoded as UTF-8, then the next line as Latin-1, then...

But the fact that a Latin-1 line decodes fine as UTF-8 doesn't mean that you get the same unicode string in both cases.

Therefore, the iterator might very well end up yielding wrong data for the line that decode fine as UTF-8.

Guessing the encoding of a file must be done for the whole file at once.